src/types/list.rs

```diff
@@ -589,6 +629,45 @@ impl<'py> BoundListIterator<'py> {
         }
     }
 
+    #[inline]
+    #[cfg(all(not(Py_LIMITED_API), feature = "nightly"))]
+    #[deny(unsafe_op_in_unsafe_fn)]
+    unsafe fn nth_back_unchecked(
+        index: &mut Index,
+        length: &mut Length,
+        list: &Bound<'py, PyList>,
+        n: usize,
+    ) -> Option<Bound<'py, PyAny>> {
+        let length_size = length.0.min(list.len());
+        if index.0 + n < length_size {
+            let target_index = length_size - n - 1;
```

Comment:
Personally this logic feels a little backwards to me, and I would probably prefer to do this using signed integers and testing if the index is less than zero. That said, this is totally equivalent and if it makes sense to you as-is then no need to change it.